source/boundary_traction/initial_lithostatic_pressure.cc

```diff
@@ -272,16 +271,29 @@ namespace aspect
     template <int dim>
     Tensor<1,dim>
     InitialLithostaticPressure<dim>::
-    boundary_traction (const types::boundary_id /*boundary_indicator*/,
+    boundary_traction (const types::boundary_id boundary_indicator,
                        const Point<dim> &position,
                        const Tensor<1,dim> &normal_vector) const
     {
       // We want to set the normal component to the vertical boundary
       // to the lithostatic pressure, the rest of the traction
       // components are left set to zero. We get the lithostatic pressure
-      // from a linear interpolation of the calculated profile.
-      Tensor<1,dim> traction;
-      traction = -interpolate_pressure(position) * normal_vector;
+      // from a linear interpolation of the calculated profile,
+      // unless boundary traction is only applied to the bottom boundary
+      // and initial topography is included.
+      // In that case we return the deepest pressure of the pressure
+      // profile directly. Otherwise, points on the bottom boundary
+      // with horizontal coordinates other than the reference point
+      // will not have a depth equal to the deepest depth of the profile.
+      const types::boundary_id bottom_bi = this->get_geometry_model().translate_symbolic_boundary_name_to_id("bottom");
+      Tensor<1, dim> traction;
+      if (!Plugins::plugin_type_matches<const InitialTopographyModel::ZeroTopography<dim>>(this->get_initial_topography_model()) &&
+          boundary_indicator == bottom_bi &&
+          traction_bi.size() == 1 &&
+          *traction_bi.begin() == bottom_bi)
```

Comment:
This whole operation is somewhat expensive (especially line 288). If you instead pre-compute the variables I mention in the header file in the `initialize` function, you can instead write here:
```suggestion
      if (prescribe_constant_pressure_at_bottom_boundary &&
          boundary_indicator == bottom_boundary_id)
```
That  would make the execution easier, and you have all necessary information in the initialize function available anyway.